op-plasma/dastate.go

```diff
@@ -124,6 +122,12 @@ func (s *State) SetInputCommitment(key []byte, committedAt uint64, challengeWind
 // initializes a new commitment and adds it to the state.
 func (s *State) GetOrTrackChallenge(key []byte, bn uint64, challengeWindow uint64) *Commitment {
 	if c, ok := s.commsByKey[string(key)]; ok {
+		// if the commitment is already tracked, mark it as canonical.
+		// and append if in order.
+		if !c.canonical {
+			s.comms = append(s.comms, c)
```

Comment:
We appear to only be pushing to `s.comms` if the commitment is canonical, but later on we check if the commitment is canonical (assuming that it may not be).